packages/noya-state/src/reducers/styleReducer.ts

```diff
@@ -280,6 +289,49 @@ export function styleReducer(
         draft.borders[index].position = position;
       });
     }
+    case 'setFillType': {
+      const [, index, type] = action;
+
+      return produce(state, (draft) => {
+        if (!draft.fills || !draft.fills[index]) return;
+        draft.fills[index].fillType = type;
+
+        if (!draft.fills[index].gradient) {
+          draft.fills[index].gradient = Models.fill.gradient;
+        }
+      });
+    }
+    case 'setGradientColor': {
```

Comment:
We'll probably want a separate `setGradientColor` and `setGradientPosition`, for more granular control. This should help with the jankiness when dragging